File: codemap/indexer/testdata/rb/lib/shop/store/repo.rb
module Shop
  module Store
    class Repo
      def save(order)
        raise NotImplementedError
      end
    end

    class PgRepo < Repo
      def self.table = 'orders'

      def save(order)
        insert(order)
      end

      def insert(order) = order
    end
  end
end

File: codemap/indexer/testdata/cpp/src/store/repo.cpp
#include "repo.hpp"

namespace acme::store {

PgRepo::PgRepo() {}

int PgRepo::save(const Order& o) { return insert(o); }

const char* PgRepo::table() { return "orders"; }

int PgRepo::insert(const Order& o) { return order_id(&o); }

}  // namespace acme::store
